src/main/java/org/zone/region/flag/meta/service/ban/flag/BanFlag.java

```diff
@@ -0,0 +1,108 @@
+package org.zone.region.flag.meta.service.ban.flag;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.zone.region.flag.Flag;
+import org.zone.region.flag.FlagType;
+import org.zone.region.flag.FlagTypes;
+import org.zone.region.flag.meta.service.ban.BanInfo;
+import org.zone.region.flag.meta.service.ban.PermanentBanInfo;
+import org.zone.region.flag.meta.service.ban.TemporaryBanInfo;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Flag for banning players from a zone
+ */
+public class BanFlag implements Flag.Serializable {
+
+    /**
+     * Collection of ban info
+     */
+    public final Collection<BanInfo> banInfo = new HashSet<>();
+
+    /**
+     * Constructor of BanFlag class
+     *
+     * @param banInfo the ban information in the form of a collection
+     */
+    public BanFlag(Collection<? extends BanInfo> banInfo) {
+        this.banInfo.addAll(banInfo);
+    }
+
+    /**
+     * Second constructor of BanFlag class that can accept a none param
+     *
+     * @param banInfo the ban information in the form of varag arrays
+     */
+    public BanFlag(BanInfo... banInfo) {
+        this(Arrays.asList(banInfo));
+    }
+
+    /**
+     * Gets the collection of ban information
+     *
+     * @return the ban info collection
+     */
+    public @NotNull Collection<BanInfo> getBanInfo() {
+        return this.banInfo;
+    }
+
+    /**
+     * Bans a player from a zone
+     *
+     * @param uuid the uuid of the player
+     *
+     * @param localDateTime the local date and time of when the player was banned (this can be null)
+     */
+    public void banPlayer(@NotNull UUID uuid, @Nullable LocalDateTime localDateTime) {
+        BanInfo banInfo;
```

Comment:
Mind adding another method

```java
public void banPlayer(@NotNull UUID uuid){
    return banPlayer(uuid, null);
}
```

This prevents people from having to lookup if the methods parameters are null, but instead can just use this method

Reply:
Ah k better

Reply:
btw u can't return a value in a void method XD

Reply:
XD